src/network.c

```diff
@@ -645,7 +616,8 @@ int write_ssl(SSL* ssl, const char* buf, int amount) {
  * @param conn the Connection
  * @param buf buffer with data to write
  * @param amount the size of the data
- * @return The amount of bytes written to the Connection
+ * @return The amount of bytes written to the Connection.
+ *         -1 when it gets an unrecoverable error, just like write().
  */
 int writen_any(Connection* conn, const void* buf, int amount) {
```

Comment:
:-/ github ate my previous draft here.

My understanding is that the busy loop bug was a result of `write()` getting what was considered a recoverable error in an infinite loop -- So, I am still concerned with this condition. When write returns -1 and errno == EINTR, write_raw is still returning '0' for recoverable error. 

So, the correct behavior of writen_any depends on correct behavior of SIGPIPE. The dependency is implicit and undocumented. The busy loop bug was a result of the implicit dependency not being respected.

I proposed two options for making this relationship not implicit: document the dependency on SIGPIPE for the exit condition, or make the dependency irrelevant by adding a retry counter. Adding a comment in the docstring is probably the easiest thing. I believe the code will be improved and clearer by doing one or the other.